Gordon360/Services/EventService.cs

```diff
@@ -115,7 +148,37 @@ public static async Task<IEnumerable<EventViewModel>> FetchEventsAsync()
             }
             catch (Exception)
             {
+                throw;
+            }
+        }
+        else
+        {
+            throw new ResourceNotFoundException();
+        }
+    }
 
+    private async Task<IEnumerable<EventViewModel>> FetchFinalExamsAsync(string finalExamsUrl)
+    {
+        using var client = new HttpClient();
+        client.Timeout = TimeSpan.FromSeconds(200);
+        client.DefaultRequestHeaders.Add("User-Agent", "Gordon360/1.0.0");
```

Comment:
Since this HttpClient is now used in two different methods, it might be better to centralize it's creation with an `HttpClientFactory`: https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory#typed-clients